service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java

```diff
@@ -347,6 +358,66 @@ public void run() {
         }
     }
 
+    // TODO: move this function to TempoService and TempoServiceImpl? (except for the publish part)
+    // TODO: break this up into smaller functions?
+    private void publishTempoSamplesToCBioPortal(Set<String> tumorPrimaryIds) throws Exception {
+        // validate and build tempo samples to publish to cBioPortal
+        Set<TempoSample> validTempoSamples = new HashSet<>();
+        for (String primaryId : tumorPrimaryIds) {
+            try {
+                // confirm sample exists by primary id
+                SmileSample sample = sampleService.getDetailedSampleByInputId(primaryId);
```

Comment:
this returns a lot of extra information that we don't really need. For the sake of just limiting how many resources we allocate to this process we can just use the single call to get tempo data by primary id and proceed (or not) from there.

```
Tempo tempo = tempoService.getTempoDataBySamplePrimaryId(primaryId);
```

Reply:
Another reason I needed the `SmileSample` instance here is to get the `cmoSampleName` (via `sample.getLatestSampleMetadata().getCmoSampleName()`) when building the protobuf object. Is there an alternative way to get `cmoSampleName`?

Reply:
Update: I'm going to add a new query to the `SampleServiceRepository` to let us query the `cmoSampleName` by `primaryId`. This will let us avoid querying the entire sample details.